### ISSUE TITLE
Debug arch file does not work with default toolchain

### ISSUE THREAD

**Issue by oschuett** [cp2k/cp2k#695]
Since #661 the generated `sdbg` arch file doen't work anymore with the default toolchain. Probably the best way to fix this is to simply not generate the debug arch files when reflapack is not present. 

Reproducer:
```
$ cd cp2k/tools/toolchain
$ ./install_cp2k_toolchain.sh
$ cd ../../arch
$ ln -s ../tools/toolchain/install/arch/* .
$ cd ..
$ make -j VERSION=sdbg

...

/usr/bin/ld: /foobar/git/cp2k/tools/toolchain/install/libxsmm-1.14/lib/libxsmm.a(libxsmm_gemm.o): in function `__real_sgemv_': libxsmm_gemm.c:(.text.__real_sgemv_+0x1): undefined reference to `sgemv_'
collect2: error: ld returned 1 exit status
```

**Comment by dev-zero** [comment]
... or use the selected BLAS/LAPACK also in the `sdbg` files

**Comment by vondele** [comment]
the reason for using reflapack in sdbg is that some tools (like valgrind) might not be able to deal with whatever techniques (new instructions, out-of-bound read-access etc) being used in optimized math libraries....

**Comment by dev-zero** [comment]
@juerghutter fyi

**Comment by oschuett** [comment]
Fixed via #827.
